contracts/test/integration/IntegrationTestVoltV2.t.sol

```diff
@@ -0,0 +1,375 @@
+//SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {DSTest} from "../unit/utils/DSTest.sol";
+import {Vm} from "../unit/utils/Vm.sol";
+import {VoltV2} from "../../volt/VoltV2.sol";
+import {Core} from "../../core/Core.sol";
+import {MainnetAddresses} from "./fixtures/MainnetAddresses.sol";
+import {stdError} from "../unit/utils/StdLib.sol";
+
+contract IntegrationTestVoltV2 is DSTest {
+    VoltV2 private volt;
+    Core private core = Core(MainnetAddresses.CORE);
+    Vm private vm = Vm(HEVM_ADDRESS);
+
+    function setUp() public {
+        volt = new VoltV2(MainnetAddresses.CORE);
+
+        vm.prank(MainnetAddresses.GOVERNOR);
+        core.grantMinter(MainnetAddresses.GOVERNOR);
+    }
+
+    function testTokenDetails() public {
+        assertEq(volt.name(), "Volt");
+        assertEq(volt.symbol(), "VOLT");
+        assertEq(volt.decimals(), 18);
+        assertEq(volt.totalSupply(), 0);
+    }
+
+    function testDelegate() public {
+        volt.delegate(address(0xFFF));
+        assertEq(volt.delegates(address(this)), address(0xFFF));
+    }
+
+    function testMintSuccessMinter(uint256 voltToMint) public {
+        vm.prank(MainnetAddresses.GOVERNOR);
+        core.grantMinter(address(this));
+
+        volt.mint(address(0xFFF), voltToMint);
+
+        assertEq(volt.totalSupply(), voltToMint);
+        assertEq(volt.balanceOf(address(0xFFF)), voltToMint);
+    }
+
+    function testMintFailureUnauthorized() public {
+        vm.expectRevert("CoreRef: Caller is not a minter");
+        volt.mint(address(0xFFF), 1e18);
+    }
+
+    function testMintFailToVoltContract() public {
+        vm.startPrank(MainnetAddresses.GOVERNOR);
+        vm.expectRevert("Volt: cannot transfer to the volt contract");
+        volt.mint(address(volt), 1e18);
+        vm.stopPrank();
+    }
+
+    function testMintFailZeroAddress() public {
+        vm.startPrank(MainnetAddresses.GOVERNOR);
+        vm.expectRevert("Volt: cannot transfer to the zero address");
+        volt.mint(address(0), 1e18);
+        vm.stopPrank();
+    }
+
+    function testBurn(uint256 voltToBurn) public {
+        vm.prank(MainnetAddresses.GOVERNOR);
+        volt.mint(address(this), voltToBurn);
+        volt.burn(voltToBurn);
```

Comment:
on burn, we need to verify the user has been properly check pointed in the voting system and has no voting rights.